learningequality/imscp
# First pass -- phase 1, extract manifest as dict, and phase 2, convert into topic tree

Works for phase 1 and 2 as described here:

https://www.notion.so/learningequality/IMS-package-Kolibri-node-s-converter-is-created-7516609da73a4f439bb0b5f9be36c108

Tested only for http://procomun.educalab.es/es/ode/view/1465806119010 for web content nodes.

#  Test plan

1. Download sample IMS package from http://procomun.educalab.es/es/ode/view/1465806119010 (find the dropdown on the top right of the page content, and select IMSCP from the list [direct link](https://agrega.educacion.es///export/es_2016062312_9100647/IMS_CP/Evento's_Solutions,_servicios_integrales_(ESSI)-IMS_CP.zip)), into the same directory as this project.
1. Uncomment `#preview_in_browser` in `ricecooker_utils.py`
1. Create a file and insert the following contents:
```
#!/usr/bin/env python

import logging
import tempfile

from ricecooker.classes import licenses

from imscp import extract_from_zip
from ricecooker_utils import make_topic_tree


if __name__ == '__main__':
    license = licenses.SpecialPermissionsLicense(
        description="XXX",
        copyright_holder="XXXX"
    )
    logging.basicConfig(level=logging.DEBUG)
    with tempfile.TemporaryDirectory() as extract_path:
        imscp_dict = extract_from_zip('eventos.zip', license, extract_path)
        for topic_dict in imscp_dict:
            topic_tree = make_topic_tree(license, topic_dict)
            print('topic tree', topic_tree)

```
1. Run it.
1. See the following

![image](https://user-images.githubusercontent.com/159687/64777774-8d587380-d58d-11e9-97d0-b3fe6a5e834d.png)

1. Comment out `preview_in_browser` in `ricecooker_utils.py`
1. Run the test runner above again, and see the following output:

```
INFO:root:Extracting zip file eventos.zip to /var/folders/z8/9d4m1_l55wl7q0ff6shs8c8w0000gn/T/tmpwp4uazxg
INFO:root:Parsing imsmanifest.xml in /var/folders/z8/9d4m1_l55wl7q0ff6shs8c8w0000gn/T/tmpwp4uazxg
INFO:root:Extracting tree structure ...
INFO:root:
topic tree Evento's Solutions, servicios integrales (ESSI) (TopicNode): 6 descendants
```
